SOURCES/COMMON.H

```diff
@@ -107,6 +107,7 @@
 
 #define	MAX_FLAGS_CUBE		80
 #define	MAX_FLAGS_GAME		255
+#define	MAX_AUX_FLAGS_CUBE	10
```

Comment:
Any reason just to be 10 max? (would using MAX_FLAGS_CUBE be enough?)
Is this to reuse a space in the save game binary file that is available?

Reply:
> Thanks for the PR @rmm17 , a lot to digest in a single pull request but I will try my best.

My intention would be to place this in a new PR but it seems Github automatically considers any additional commits on "main" on my fork to be part of this PR :D. Maybe there's a way to avoid this

> Had a quick looks through all the code. It seems all ok. I wonder why you use AuxCubeFlags instead of the actual scene Cube Flags?
> 
> There were quite a few changes in the scripting behaviour because of the save game, I wonder if this could have been avoided since this changes may introduce scripting bugs.

I have added this to avoid a few scenarios where NPCs drop more than one key when I reload a save (the grobo in Funfrock HQ and the fisherman in Tippet Island). You could think of these flags as extra to the ones that already exist in ListCubeFlag (and are also persisted). I've done this to avoid changing the actual ListCubeFlag, so in a way this was to avoid introducing additional bugs by messing with flags which I don't know whether they are or are not being used for other purposes, but maybe I'm being too cautious on this and there may be room for improvement here.

> Any reason just to be 10 max? (would using MAX_FLAGS_CUBE be enough?) Is this to reuse a space in the save game binary file that is available?

Because I'm only using the aux flags in 2 scenarios in 2 different cubes right now. In order to not occupy much more space in the save file, I felt more wouldn't be necessary. The max number is also placed in the save, so incrementing it should be supported, but I doubt there will be many additional scenarios where this will be used to justify incrementing it right now.

Reply:
> My intention would be to place this in a new PR but it seems Github automatically considers any additional commits on "main" on my fork to be part of this PR :D. Maybe there's a way to avoid this

The best way is always to work on a separate feature branch on your local repository and when you push it into your remote git repo, you have the option to create the PR in the upstream repo.

> I have added this to avoid a few scenarios where NPCs drop more than one key when I reload a save (the grobo in Funfrock HQ and the fisherman in Tippet Island)... 

I understood the reason, I just thought the same effect could be achieved with Cube Flags, as long as they were persisted in the save game.